From d8f1b9fd7df059daa7f5f6e24165089cec6e206d Mon Sep 17 00:00:00 2001
From: "Tan, Raymond" <raymond.tan@intel.com>
Date: Thu, 2 Jul 2015 08:25:40 +0800
Subject: [PATCH 337/389] net: stmmac: Add module param to enable/disable PCI
 MSI mode

Added the module param to allow user to enable/disable PCI MSI mode
on stmmac-pci driver. It's enabled to MSI mode by default on Quark
platform.

Signed-off-by: Tan, Raymond <raymond.tan@intel.com>
Signed-off-by: Dheeraj Jamwal <dheerajx.s.jamwal@intel.com>
Change-Id: Ie3e4abc9becbf917c0c67b5cf3cb20a400eeb433
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/5511
Reviewed-by: Fung, Joshua <joshua.fung@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
index 85412ad..428261f 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_pci.c
@@ -28,6 +28,10 @@
 
 #include "stmmac.h"
 
+static int enable_msi = 1;
+module_param(enable_msi, int, S_IRUGO | S_IWUSR);
+MODULE_PARM_DESC(enable_msi, "Enable PCI MSI mode");
+
 /*
  * This struct is used to associate PCI Function of MAC controller on a board,
  * discovered via DMI, with the address of PHY connected to the MAC. The
@@ -262,7 +266,13 @@ static int stmmac_pci_probe(struct pci_dev *pdev,
 	} else
 		stmmac_default_data(plat);
 
-	pci_enable_msi(pdev);
+	if (enable_msi == 1) {
+		ret = pci_enable_msi(pdev);
+		if (ret)
+			dev_info(&pdev->dev, "stmmac MSI mode NOT enabled\n");
+		else
+			dev_info(&pdev->dev, "stmmac MSI mode enabled\n");
+	}
 
 	priv = stmmac_dvr_probe(&pdev->dev, plat, pcim_iomap_table(pdev)[i]);
 	if (IS_ERR(priv)) {
-- 
1.9.3

